scripts/build/Jenkins/Jenkinsfile

```diff
@@ -384,49 +384,49 @@ def NoControlFile(repoName, processed_repos, o3de_project, defer_enable_gems) {
 
         //a singular object repo will have one of these o3de json files in the root
         if(fileExists("engine.json")) {
-            PlatformSh("${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} register --engine-path ${workspace}/${repoName} || dir>nul", "Registering engine in ${workspace}/${repoName}")
-        }
-        else if(fileExists("project.json")) {
-            PlatformSh("${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} register --project-path ${workspace}/${repoName} || dir>nul", "Registering project in ${workspace}/${repoName}")
+            PlatformSh("${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} register --engine-path ${workspace}/${repoName} || echo>nul", "Registering engine in ${workspace}/${repoName}")
         }
         else if(fileExists("gem.json")) {
-            PlatformSh("${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} register --gem-path ${workspace}/${repoName} || dir>nul", "Registering gem in ${workspace}/${repoName}")
-            defer_enable_gems.add([COMMAND: "${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} enable-gem --gem-path ${workspace}/${repoName} --project-name ${o3de_project} || dir>nul", ECHO: "Enabling ${gemName} in ${o3de_project}"])
+            PlatformSh("${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} register --gem-path ${workspace}/${repoName} || echo>nul", "Registering gem in ${workspace}/${repoName}")
+            defer_enable_gems.add([COMMAND: "${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} enable-gem --gem-path ${workspace}/${repoName} --project-name ${o3de_project} || echo>nul", ECHO: "Enabling ${gemName} in ${o3de_project}"])
```

Comment:
A Gem can be enabled with the project using it's path via the `--project-path` argument.
In that case the project doesn't have to be registered in order to activate the gem.
Registration is needed to query the project path via using the "project name"

Reply:
Correct, this is more to resolve issues where the project is registered before a required gem, which results in a unsatisfied dependency error message